src/metabase/api/search.clj

```diff
@@ -216,6 +216,17 @@
   {:arglists '([model search-context])}
   (fn [model _] model))
 
+(defn collection-id-for-table
+  "What column should we use to a) join to the `collection` table? and b) filter collections to the set the user is
+  allowed to see?
+
+  When we're not looking for collections, use `{my-table}.collection_id` or `{my-table}.trashed_from_collection_id`,
+  depending on if we're looking for archived items."
```

Comment:
I think I get what this is going for — `trashed_from_collection_id` will only be non-nil when something is in the trash, so we always can use it if it's set, and otherwise use `collection_id`.

But the wording here is a bit confusing without that knowledge. It seems to be implying that this function should be returning something different depending on the input. Maybe you could clarify that a bit.